cdc/http_handler.go

```diff
@@ -65,8 +65,10 @@ func (s *Server) handleResignOwner(w http.ResponseWriter, req *http.Request) {
 		writeError(w, http.StatusBadRequest, errors.New("this api only supports POST method"))
 		return
 	}
+	s.ownerLock.RLock()
 	if s.owner == nil {
 		handleOwnerResp(w, concurrency.ErrElectionNoLeader)
```

Comment:
why not put the operations of `lock\unlock` and checking nil into handleOwnerResp function

Reply:
bad suggest, ignore me